bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolBridgeHandler.java

```diff
@@ -0,0 +1,472 @@
+                                                 * not (yet) transmitting it
+                                                 */
+                                            }
+                                            break;
+                                        case DateTime:
+                                            thingHandler.setChannelValue(channelId, pfv.getRawValueDate());
+                                            break;
+                                        case WeekTime:
+                                        case Time:
+                                        default:
+                                            thingHandler.setChannelValue(channelId,
+                                                    pfv.formatTextValue(pfv.getPacketFieldSpec().getUnit(), locale));
+                                    }
+
+                                    if (pfv.getEnumVariant() != null) {
+                                        // if we have an enum, we additionally add that as channel
+                                        String enumChannelId = channelId + "-str";
+                                        if (thing.getChannel(enumChannelId) == null) {
+                                            ChannelTypeUID enumChannelTypeUID = new ChannelTypeUID(
+                                                    ResolBindingConstants.BINDING_ID, "None");
+                                            ThingBuilder thingBuilder = thingHandler.editThing();
+
+                                            ChannelUID enumChannelUID = new ChannelUID(thing.getUID(), enumChannelId);
+                                            Channel channel = ChannelBuilder.create(enumChannelUID, "String")
+                                                    .withType(enumChannelTypeUID).withLabel(pfv.getName(lang)).build();
+
+                                            thingBuilder.withChannel(channel).withLabel(thing.getLabel());
+
+                                            thingHandler.updateThing(thingBuilder.build());
+                                        }
+
+                                        thingHandler.setChannelValue(enumChannelId, pfv.getEnumVariant().getText(lang));
+                                    }
+                                } else {
+                                    logger.trace("ThingHandler for {} not registered.", thingType);
+                                }
+                            }
+                        }
+                    });
+
+                    // Establish the connection
+                    tcpConnection.connect();
+                    Thread.sleep(1000); // after a reconnect wait 1 sec
+                    isConnected = (tcpConnection.getConnectionState().equals(Connection.ConnectionState.CONNECTED));
+                } catch (IOException e) {
+                    logger.trace("Connection failed", e);
```

Comment:
Better include the exception's message and remove logging the stack trace. As logging the stack trace, when the network fails seems a bit unreasonable.